protocols/gossipsub/src/handler.rs

```diff
@@ -571,10 +529,37 @@ impl ConnectionHandler for Handler {
             ConnectionEvent::FullyNegotiatedOutbound(fully_negotiated_outbound) => {
                 self.on_fully_negotiated_outbound(fully_negotiated_outbound)
             }
-            ConnectionEvent::DialUpgradeError(DialUpgradeError { error: e, .. }) => {
+            ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
                 self.outbound_substream_establishing = false;
-                warn!("Dial upgrade error {:?}", e);
-                self.upgrade_errors.push_back(e);
+
+                match error {
+                    // Timeout errors get mapped to NegotiationTimeout and we close the connection.
+                    ConnectionHandlerUpgrErr::Timeout | ConnectionHandlerUpgrErr::Timer => {
+                        self.keep_alive = KeepAlive::No;
+                        log::info!("Dial upgrade error: {}", HandlerError::NegotiationTimeout);
```

Comment:
Here too, no need for the `HandlerError` indirection if we can just log it directly.

Reply:
I think we can deprecate the entire `HandlerError` type.

Reply:
Unfortunately, no. Variants `MaxTransmissionSize` and `Codec` are used when implementing, e.g., `impl<TSocket> InboundUpgrade<TSocket> for ProtocolConfig`.
I have removed other variants.